examples/standalone/runfile/dynamics.py

```diff
@@ -0,0 +1,142 @@
+from argparse import ArgumentParser
+
+import mpi4py
+import serialbox
+import yaml
+
+import fv3core
+import fv3core._config as spec
+import fv3core.stencils.fv_dynamics as fv_dynamics
+import fv3core.testing
+import fv3gfs.util as util
+from fv3core.testing import timer, write_to_json
+
+
+if __name__ == "__main__":
+    timing = timer()
+    timing.time("init")
+
+    usage = "usage: python %(prog)s <data_dir> <timesteps> <backend>"
+    parser = ArgumentParser(usage=usage)
+
+    parser.add_argument(
+        "data_dir",
+        type=str,
+        action="store",
+        help="directory containing data to run with",
+    )
+    parser.add_argument(
+        "time_step",
+        type=int,
+        action="store",
+        help="number of timesteps to execute",
+    )
+    parser.add_argument(
+        "backend",
+        type=str,
+        action="store",
+        help="path to the namelist",
+    )
+    args = parser.parse_args()
+    backend = args.backend
+    data_dir = args.data_dir
+    time_step = args.time_step
+
+    # # MPI stuff
+    comm = mpi4py.MPI.COMM_WORLD
+    rank = comm.Get_rank()
+
+    # fv3core specific setup
+    fv3core.set_backend(backend)
+    fv3core.set_rebuild(False)
+
+    # namelist setup
+    spec.set_namelist(data_dir + "/input.nml")
+
+    nml2 = yaml.safe_load(
+        open(
+            data_dir + "/input.yml",
+            "r",
+        )
+    )["namelist"]
+    experiment_name = yaml.safe_load(
+        open(
+            data_dir + "/input.yml",
+            "r",
+        )
+    )["experiment_name"]
+
+    # set up of helper structures
+    serializer = serialbox.Serializer(
+        serialbox.OpenModeKind.Read,
+        data_dir,
+        "Generator_rank" + str(rank),
+    )
+    cube_comm = util.CubedSphereCommunicator(
+        comm, util.CubedSpherePartitioner.from_namelist(nml2)
+    )
+
+    # get grid from serialized data
+    grid_savepoint = serializer.get_savepoint("Grid-Info")[0]
+    grid_data = {}
+    grid_fields = serializer.fields_at_savepoint(grid_savepoint)
+    for field in grid_fields:
+        grid_data[field] = serializer.read(field, grid_savepoint)
+        if len(grid_data[field].flatten()) == 1:
+            grid_data[field] = grid_data[field][0]
+    grid = fv3core.testing.TranslateGrid(grid_data, rank).python_grid()
+    spec.set_grid(grid)
```

Comment:
might be helpful to put into a separate method, but I can see arguments for keeping it inlined

Reply:
I don't really see a benefit of abstracting it away here

Reply:
FWIW I think it's fine in a scenario like this (a runfile example) to have code in the long-block-of-text style. But I thought I should point out some of the benefits of what @rheacangeo suggested if it's not clear. These benefits come into play in most coding since it's related to the rule of thumb that you should keep functions short.

One main benefit is if you do it to all the code, it makes it easier to both scan the main block to tell what the script does, and understand the information flow between tasks. e.g. (of course this won't run):
```
args = parse_args()
config = load_config(args)
cube = get_communicator(config["layout"))
state, grid = initialize_state_from_savepoint(config["data_dir"])
fv_dynamics.fv_dynamics(state, ...)
...
```
Doing things like this would also make it very clear e.g. what information is actually necessary to define the grid? The answer is either data_dir or serializer, but it takes some time reading the code to figure that out. When you notice yourself writing a lot of comments to describe the parts of a function (or main block), it's a cue that you may want to move each of those blocks into another named function. This has the added benefit of making it less likely those comments will go out of date.